pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,326 @@
+							ip.String(), 1)
+						log.Functionf(
+							"prepareLocalProfileServerMap: will use %s for bridge %s",
+							localServerURLReplaced, ulStatus.Bridge)
+						appendURLAndSrcIPMap(res, ulStatus.Bridge,
+							&urlAndSrcIP{actualURL: localServerURLReplaced, srcIP: bridgeIP})
+					}
+				}
+			}
+		}
+	}
+	return res, nil
+}
+
+// getLocalProfileConfig connects to local profile server to fetch current profile
+func getLocalProfileConfig(localServerURL string, getconfigCtx *getconfigContext) (*profile.LocalProfile, error) {
+
+	log.Functionf("getLocalProfileConfig(%s)", localServerURL)
+
+	localServerMap, err := prepareLocalProfileServerMap(localServerURL, getconfigCtx)
+	if err != nil {
+		return nil, fmt.Errorf("getLocalProfileConfig: prepareLocalProfileServerMap: %s", err)
+	}
+
+	if len(localServerMap) == 0 {
+		return nil, fmt.Errorf(
+			"getLocalProfileConfig: cannot find any configured apps for localServerURL: %s",
+			localServerURL)
+	}
+
+	var errList []string
+	for bridgeName, urlAndSrcIPs := range localServerMap {
+		for _, el := range urlAndSrcIPs {
+			resp, contents, err := zedcloud.SendLocal(zedcloudCtx, el.actualURL, bridgeName, el.srcIP, 0, nil)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+				continue
+			}
+			if resp.StatusCode != http.StatusOK {
+				errList = append(errList, fmt.Sprintf("SendLocal: wrong response status code: %d",
+					resp.StatusCode))
+				continue
+			}
+			if err := validateProtoMessage(el.actualURL, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
+			localProfile, err := parseAndValidateLocalProfile(contents, getconfigCtx)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("parseAndValidateLocalProfile: %s", err))
+				continue
+			}
+			return localProfile, nil
+		}
+	}
+	return nil, fmt.Errorf("getLocalProfileConfig: all attempts failed: %s", strings.Join(errList, ";"))
+}
+
+func writeReceivedProtoMessageLocalProfile(localProfile *profile.LocalProfile) {
+	contents, err := proto.Marshal(localProfile)
+	if err != nil {
+		log.Errorf("writeReceivedProtoMessageLocalProfile Marshalling failed: %s", err)
+		return
+	}
+	writeProtoMessage(savedLocalProfileFile, contents)
+	return
+}
+
+//parseProfile process local and global profile configuration
+//must be called before processing of app instances from config
+func parseProfile(ctx *getconfigContext, config *zconfig.EdgeDevConfig) {
+	log.Functionf("parseProfile start: globalProfile: %s localProfile: %s",
+		ctx.globalProfile, ctx.localProfile)
+	if ctx.globalProfile != config.GlobalProfile {
+		log.Noticef("parseProfile: GlobalProfile changed from %s to %s",
+			ctx.globalProfile, config.GlobalProfile)
+		ctx.globalProfile = config.GlobalProfile
+	}
+	if ctx.localProfileServer != config.LocalProfileServer {
+		// read saved config on the first call
```

Comment:
No, I don't want this complexity here. Just set ctx.localProfileServer and trigger.